src/containers/dynamicarray.d

```diff
@@ -119,6 +119,22 @@ struct DynamicArray(T, Allocator = Mallocator, bool supportGC = shouldAddGCRange
 				GC.addRange(arr.ptr, arr.length * T.sizeof);
 			}
 		}
+		static if (is(T == struct))
+		{
+			// If a destructor is run before blit or copying involves
+			// more than just a blit, ensure that arr[l] is in a valid
+			// state before assigning to it.
+			import std.traits: hasElaborateAssign, hasElaborateDestructor;
+			static if (hasElaborateAssign!T || hasElaborateDestructor!T)
```

Comment:
Can this be merged with the `static if` above?

```d
static if (is(T == struct) && (hasElaborateAssign!T || hasElaborateDestructor!T))
```

Reply:
sure